app/grandchallenge/codebuild/client.py

```diff
@@ -0,0 +1,99 @@
+import boto3
+from django.conf import settings
+from django.core import files
+from django.core.files.temp import NamedTemporaryFile
+
+from grandchallenge.algorithms.models import AlgorithmImage
+from grandchallenge.codebuild.models import Build
+from grandchallenge.core.storage import private_s3_storage
+
+
+class CodeBuildClient:
+    build = None
+
+    def __init__(
+        self, *, project_name=None, msg=None, algorithm=None, build_id=None,
+    ):
+        self.client = boto3.client(
+            "codebuild",
+            aws_access_key_id=settings.CODEBUILD_ACCESS_KEY,
+            aws_secret_access_key=settings.CODEBUILD_SECRET_KEY,
+            region_name=settings.CODEBUILD_REGION,
+        )
+        self.log_client = boto3.client(
+            "logs",
+            aws_access_key_id=settings.CODEBUILD_ACCESS_KEY,
+            aws_secret_access_key=settings.CODEBUILD_SECRET_KEY,
+            region_name=settings.CODEBUILD_REGION,
+        )
+        self.project_name = project_name
+        self.msg = msg
+        self.algorithm = algorithm
+        self.build_id = build_id
+        self.config = {
+            "name": project_name,
+            "source": {"type": "S3", "location": ""},
+            "artifacts": {
+                "type": "S3",
+                "location": settings.PRIVATE_S3_STORAGE_KWARGS["bucket_name"],
+            },
+            "environment": {
+                "type": "LINUX_CONTAINER",
+                "image": "docker:dind",
+                "computeType": "BUILD_GENERAL1_SMALL",
+                "privilegedMode": True,
+                "registryCredential": {
+                    "credential": settings.DOCKER_SECRET_ARN,
+                    "credentialProvider": "SECRETS_MANAGER",
+                },
+                "imagePullCredentialsType": "SERVICE_ROLE",
+            },
+            "serviceRole": settings.CODEBUILD_SERVICE_ROLE,
+            "encryptionKey": settings.CODEBUILD_ENCRYPTION_KEY,
+        }
+
+    def create_build_project(self, *, source):
+        self.config["source"]["location"] = source
+        self.client.create_project(**self.config)
```

Comment:
This creates a new project for each build, right? Would it not make more sense for use to create the project in Terraform, and then we use it from grand challenge with some parametrisation eg with the input and output variables? Is that even possible?

Reply:
No unfortunately not. The source file location is set on build project creation, so we need a new project for each new zip file. I could look into a workaround for this in the cooldown week, but I think this is at least the official way to do things.